### PR TITLE
[image_picker] fix crash when aar from 'flutter build aar'

### PR DESCRIPTION
**Pull request by liujingtech**
## Description
Use context cast application in ImagePickerPlugin#onCreate,when depend module context type is application, but depend on aar context type is acitivity, finally cause ClassCastException.

## Related Issues

[flutter/flutter#41558](https://github.com/flutter/flutter/issues/41558)

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

### CODE REVIEW

**Comment by liujingtech** [comment]
Why this [task](https://github.com/flutter/plugins/pull/2120/checks?check_run_id=242495249) still running

**Comment by liujingtech** [comment]
> The fix LGTM! Can we add some tests for this change?

done.

File: packages/image_picker/example/android/app/src/test/java/io/flutter/plugins/imagepicker/ImagePickerPluginTest.java [hunk 1]
```diff
@@ -109,6 +109,15 @@ public void onResiter_WhenAcitivityIsNull_ShouldNotCrash() {
         "No exception thrown when ImagePickerPlugin.registerWith ran with activity = null", true);
   }
 
+  @Test
+  public void onConstructor_WhenContextTypeIsActivity_ShouldNotCrash() {
+    // When you use Module to dependency, the type of Context will be Application.but when you use AAR to dependency on this plugin, the type of Context will be Activity.
```
Comment (cyanglaz) [inline packages/image_picker/example/android/app/src/test/java/io/flutter/plugins/imagepicker/ImagePickerPluginTest.java]: Nits: The test case name is pretty self explanatory. We can remove this comments to reduce some noise.

File: packages/image_picker/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+## 0.6.1+7
+
+* Android: Fix ImagePickerPlugin#onCreate cast context may be cause exeception.
```
Comment (cyanglaz) [inline packages/image_picker/CHANGELOG.md]: nits:
Android: Fix ImagePickerPlugin#onCreate casting context which causes exception.

**Comment by liujingtech** [comment]
Can I know what else I need to do to help the code merge into the master branch? @cyanglaz

**Comment by cyanglaz** [comment]
@liujingtech Could you update pubspec.ymal with the new version

**Comment by liujingtech** [comment]
> @liujingtech Could you update pubspec.ymal with the new version

Sorry, I forgot to update pubspec.yaml

**Comment by liujingtech** [comment]
> @liujingtech Could you update pubspec.ymal with the new version

Hi @cyanglaz , What else I can do ?

**Pull request merged**